src/components/common/Directors.vue

```diff
@@ -1325,128 +1337,105 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
    * @param director The director to check.
    * @returns Whether the director is actionable.
    */
-  private isActionable (director): boolean {
-    return director.isDirectorActionable !== undefined ? director.isDirectorActionable : true
+  private isActionable (director: DirectorIF): boolean {
+    return (director.isDirectorActionable !== undefined) ? director.isDirectorActionable : true
   }
 
-  /**
-   * If we have paid director changes (add or cease) add a single fee to the filing.
-   * - when we've made one change, add the fee
-   * - when we've removed/undone all changes, remove the fee
-   */
-  @Watch('directorsChange')
+  /** Called when there is a change (ie, some <-> none) in directors with a paid change. */
+  @Watch('directorsPaidChange')
   private onDirectorsChange (val: boolean): void {
     // emit event back up to parent
     this.emitDirectorsChange(val)
   }
 
-  /**
-   * If we have free director changes (add or cease) add a single free fee code to the filing.
-   */
+  /** Called when there is a change (ie, some <-> none) in directors with a free change. */
   @Watch('directorsFreeChange')
   private onDirectorsFreeChange (val: boolean): void {
     // emit event back up to parent
     this.emitDirectorsFreeChange(val)
   }
 
-  /**
-   * When a director form's validity changes, inform parent component.
-   */
+  /** Called when a director form's validity changes. */
   @Watch('directorFormValid')
   private onDirectorFormValid (val: boolean): void {
+    // emit event back up to parent
     this.emitDirectorFormValid(val)
   }
 
-  /**
-   * When as-of date is initialized, set initial director appointment date.
-   */
-  @Watch('asOfDate', { immediate: true })
-  private onAsOfDateInitialized (date: string): void {
+  /** Called when As Of Date is set or changes (from parent component). */
+  @Watch('asOfDate')
+  private async onAsOfDateChange (date: string): Promise<void> {
+    // set director appoinment date for new directors
     this.director.appointmentDate = date
```

Comment:
This did nothing in the normal code because As Of Date is initialized to null.

This was used in some unit tests where Prop Data was set when mounting this component. But it's a bad idea for the tests to run differently from normal code.